test/Validate.Capability.cpp

```diff
@@ -0,0 +1,428 @@
+                                                           )));
+
+INSTANTIATE_TEST_CASE_P(OpVariable,
+                        ValidateCapability,
+                        ::testing::Combine(
+testing::ValuesIn(kAllCapabilities),
+testing::Values(
+make_pair(" %intt = OpTypeInt 32 0\n"
+          " %ptrt = OpTypePointer UniformConstant %intt\n"
+          " %var = OpVariable %ptrt UniformConstant\n",             kAllCapabilities),
+make_pair(" %intt = OpTypeInt 32 0\n"
+          " %ptrt = OpTypePointer Input %intt"
+          " %var = OpVariable %ptrt Input\n",                       kShaderDependencies),
+make_pair(" %intt = OpTypeInt 32 0\n"
+          " %ptrt = OpTypePointer Uniform %intt\n"
+          " %var = OpVariable %ptrt Uniform\n",                     kShaderDependencies),
+make_pair(" %intt = OpTypeInt 32 0\n"
+          " %ptrt = OpTypePointer Output %intt\n"
+          " %var = OpVariable %ptrt Output\n",                      kShaderDependencies),
+make_pair(" %intt = OpTypeInt 32 0\n"
+          " %ptrt = OpTypePointer Workgroup %intt\n"
+          " %var = OpVariable %ptrt Workgroup\n",                   kAllCapabilities),
+make_pair(" %intt = OpTypeInt 32 0\n"
+          " %ptrt = OpTypePointer CrossWorkgroup %intt\n"
+          " %var = OpVariable %ptrt CrossWorkgroup\n",              kAllCapabilities),
+make_pair(" %intt = OpTypeInt 32 0\n"
+          " %ptrt = OpTypePointer Private %intt\n"
+          " %var = OpVariable %ptrt Private\n",                     kShaderDependencies),
+// TODO(umar): OpVariable Function storage class
+make_pair(" %intt = OpTypeInt 32 0\n"
+          " %ptrt = OpTypePointer Generic %intt\n"
+          " %var = OpVariable %ptrt Generic\n",                     kKernelDependencies),
+make_pair(" %intt = OpTypeInt 32 0\n"
+          " %ptrt = OpTypePointer PushConstant %intt\n"
+          " %var = OpVariable %ptrt PushConstant\n",                kShaderDependencies),
+make_pair(" %intt = OpTypeInt 32 0\n"
+          " %ptrt = OpTypePointer AtomicCounter %intt\n"
+          " %var = OpVariable %ptrt AtomicCounter\n",               vector<string>{"AtomicStorage"}),
+make_pair(" %intt = OpTypeInt 32 0\n"
+          " %ptrt = OpTypePointer Image %intt\n"
+          " %var = OpVariable %ptrt Image\n",                       kAllCapabilities)
+  )));
+// clang-format on
+
+//TODO(umar): All other instructions
+
+TEST_P(ValidateCapability, Capability) {
+  string capability;
+  pair<string, vector<string>> operation;
+  std::tie(capability, operation) = GetParam();
+  stringstream ss;
+  if(capability.empty() == false) {
+    ss << "OpCapability " + capability;
+  }
+
+  ss << "\nOpMemoryModel Logical GLSL450\n"
+     << operation.first;
+
+  spv_result_t res = SPV_ERROR_INTERNAL;
+  auto& valid_capabilities = operation.second;
+
+  auto it = find(begin(valid_capabilities), end(valid_capabilities), capability);
+  if(it != end(operation.second)) {
```

Comment:
Nit: I'd use `valid_capabilities` here instead of `operation.second`, for clarity.